functions/utils/getRequest.test.ts

```diff
@@ -0,0 +1,37 @@
+import * as matchers from 'jest-extended'
+expect.extend(matchers)
+
+import Cache from './cache'
+import getRequest from './getRequest'
+
+jest.mock('./cache', () => ({
+  match: jest.fn(),
+  put: jest.fn(),
+}))
+
+test('should call Cache.match before calling getData when request is not cached', async () => {
+  const url = 'https://example.com'
+  const getData = jest.fn().mockResolvedValueOnce({
+    title: 'test',
+    image: 'testImage',
+    url: 'testUrl',
+  })
+  await getRequest(Promise.resolve(new Response()), url, getData)
+  expect(Cache.match).toHaveBeenCalledWith(url)
+  expect(getData).toHaveBeenCalled()
+  expect(Cache.match).toHaveBeenCalledBefore(getData)
+  expect(Cache.put).toHaveBeenCalledAfter(getData)
+})
+
+test('getData should not be called when request is cached', async () => {
+  const url = 'https://example.com'
+  ;(Cache.match as jest.Mock).mockResolvedValueOnce({
```

Comment:
```suggestion
  jest.mocked(Cache.match).mockResolvedValueOnce({
```

or if `jest.mocked` is not available (because we may be on an old version of jest) use the `mocked` (or copy it over) from our test-utils.